pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -792,9 +960,20 @@ private void cleanUpDelayMetrics(List<TableConfig> tableConfigs) {
       Map<String, String> taskConfigs = currentTableConfig.getTaskConfig().getConfigsForTaskType(getTaskType());
       Map<String, Map<String, String>> mergeLevelToConfigs = MergeRollupTaskUtils.getLevelToConfigMap(taskConfigs);
       Map<String, Long> tableToWatermark = _mergeRollupWatermarks.get(tableNameWithType);
-      for (String mergeLevel : tableToWatermark.keySet()) {
-        if (!mergeLevelToConfigs.containsKey(mergeLevel)) {
-          resetDelayMetrics(tableNameWithType, mergeLevel);
+      if (tableToWatermark != null) {
```

Comment:
`_mergeRollupWatermarks` map is used to emit the old `mergeRollupTaskDelayInNumBuckets` metric and will be set for the default mode. The new `processAll` mode does not emit the old metric, and won't set this map, so this can be null.